### PR TITLE
handle special address

### PR DESCRIPTION
**Pull request by dustinxie**

### CODE REVIEW

File: address/address_special.go [hunk 1]
```diff
@@ -52,3 +80,38 @@ func bytesToHash160(b []byte) Hash160 {
...
+func newAddrV1Special(s string) *AddrV1Special {
+	return &AddrV1Special{
+		addr: s,
+	}
+}
+
+// IsAddrV1Special returns true for special address
+func IsAddrV1Special(s string) bool {
+	switch s {
+	case RewardingPoolAddr, StakingBucketPoolAddr:
```
[the remark this answers is not in the document]
Comment (dustinxie) [reply address/address_special.go:99]: no, those are valid bech32 encoding, can be converted between 20-byte hash and string, like
```
[]byte("stakingCreate") <--> "io1qqqqqqqqqqq8xarpdd5kue6rwfjkzar9k0wk6t"
```

File: address/address_v1.go [hunk 2]
```diff
@@ -45,13 +48,15 @@ func (v *v1) FromBytes(bytes []byte) (*AddrV1, error) {
 }
 
 // FromHex converts a hex-encoded string into an address struct
-func (v *v1) FromHex(s string) (*AddrV1, error) {
+func (v *v1) FromHex(s string) (Address, error) {
 	if len(s) > 1 {
-		if s[:2] == "0x" || s[:2] == "0X" {
+		if s[0] == '0' && (s[1] == 'x' || s[1] == 'X') {
```
[the remark this answers is not in the document]
Comment (dustinxie) [reply address/address_v1.go:53]: this is not iotex-core repo

File: address/address_v1.go [hunk 3]
```diff
@@ -45,13 +48,15 @@ func (v *v1) FromBytes(bytes []byte) (*AddrV1, error) {
...
 			s = s[2:]
 		}
 	}
-
+	if len(s)%2 == 1 {
+		s = "0" + s
```
[the remark this answers is not in the document]
Comment (dustinxie) [reply address/address_v1.go:58]: checked and made it compatible with ethereum: https://github.com/ethereum/go-ethereum/blob/master/common/bytes.go#L30-L32

so if input is missing a leading 0, it still works

File: address/address_v1.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (dustinxie) [reply address/address_v1.go:53]: if later same code come up again, can make a `util.Remove0xPrefix`

**Review by Liuhaai — APPROVED** [review]

**Pull request merged**